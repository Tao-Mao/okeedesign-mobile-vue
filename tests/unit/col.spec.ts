import { mount } from '../index';
import { Col as ColNative, Row as RowNative } from '@src';
const Col = ColNative as any;
const Row = RowNative as any;

describe('Col', () => {
  test('gutter prop', () => {
    const wrapper = mount({
      template: `
      <o-row gutter="24">
        <o-col span="24">24</o-col>

        <o-col span="12">12</o-col>
        <o-col span="12">12</o-col>

        <o-col span="8">8</o-col>
        <o-col span="8">8</o-col>
        <o-col span="8">8</o-col>

        <o-col span="6">6</o-col>
        <o-col span="6">6</o-col>
        <o-col span="6">6</o-col>
        <o-col span="6">6</o-col>
        
        <o-col span="7">7</o-col>
        <o-col span="6">6</o-col>
        <o-col span="5">5</o-col>
        <o-col span="4">4</o-col>
        <o-col span="3">3</o-col>
        <o-col span="2">2</o-col>
      </o-row>
    `,
      components: {
        'o-col': Col,
        'o-row': Row,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
