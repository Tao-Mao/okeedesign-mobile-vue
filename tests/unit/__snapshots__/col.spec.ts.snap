// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col gutter prop 1`] = `
<div class="o-row" style="margin-left: -12px; margin-right: -12px;">
  <div class="o-col o-col--24" style="padding-left: 12px; padding-right: 12px;">24</div>
  <div class="o-col o-col--12" style="padding-left: 12px; padding-right: 12px;">12</div>
  <div class="o-col o-col--12" style="padding-left: 12px; padding-right: 12px;">12</div>
  <div class="o-col o-col--8" style="padding-left: 12px; padding-right: 12px;">8</div>
  <div class="o-col o-col--8" style="padding-left: 12px; padding-right: 12px;">8</div>
  <div class="o-col o-col--8" style="padding-left: 12px; padding-right: 12px;">8</div>
  <div class="o-col o-col--6" style="padding-left: 12px; padding-right: 12px;">6</div>
  <div class="o-col o-col--6" style="padding-left: 12px; padding-right: 12px;">6</div>
  <div class="o-col o-col--6" style="padding-left: 12px; padding-right: 12px;">6</div>
  <div class="o-col o-col--6" style="padding-left: 12px; padding-right: 12px;">6</div>
  <div class="o-col o-col--7" style="padding-left: 12px; padding-right: 12px;">7</div>
  <div class="o-col o-col--6" style="padding-left: 12px; padding-right: 12px;">6</div>
  <div class="o-col o-col--5" style="padding-left: 12px; padding-right: 12px;">5</div>
  <div class="o-col o-col--4" style="padding-left: 12px; padding-right: 12px;">4</div>
  <div class="o-col o-col--3" style="padding-left: 12px; padding-right: 12px;">3</div>
  <div class="o-col o-col--2" style="padding-left: 12px; padding-right: 12px;">2</div>
</div>
`;
